src/platform/forms-system/src/js/containers/FormPage.jsx

```diff
@@ -29,6 +29,19 @@ const scrollToTop = () => {
 };
 
 class FormPage extends React.Component {
+  constructor(props) {
+    super(props);
+    // If the page config has an ancillaryData property, connect to the redux store and get the data
+    //  it specifies.
+    // Setting it here so SchemaForm doesn't get re-mounted a bunch
```

Comment:
Does this make enough sense?

Reply:
FormPage is already connected to the Redux store, I think you could put the ancillaryData selector in `mapStateToProps` and then pass it to SchemaForm directly.

Also, you'll need to map this the other places that use SchemaForm, like ReviewCollapsibleChapter (or it's connected parent) and ArrayField. If this mapping is too complex, we might have to just connect SchemaForm.

Reply:
Ahh, right. `mapStateToProps` has an `ownProps` parameter that I completely forgot about. That's much more elegant, innit? :+1: